lib/middleware/validateSession.js

```diff
@@ -0,0 +1,63 @@
+var getSession = function (req, key) {
+  return key.split(".").reduce(function(res, key) {
```

Comment:
I don't think reduce would work here.

You could have a function that iterates through the keys and returns the found key or null

You could try something like http://stackoverflow.com/questions/6491463/accessing-nested-javascript-objects-with-string-key

Alternatively you can assume that if the sessionKey is not in the body, then it is in the header so you could have a 

```
opts.sessionHeader='x-fh-some-session-header'
```